vowpalwabbit/core/src/reductions/automl.cc

```diff
@@ -92,16 +92,49 @@ void finish_example(VW::workspace& all, automl<CMType>& data, VW::multi_ex& ec)
   VW::finish_example(all, ec);
 }
 
+template <typename CMType>
+void output_single_model(VW::workspace& all, automl<CMType>& data)
+{
+  options_i& options = *all.options;
+  if (!options.was_supplied("predict_only_model")) { return; }
+  // Clear non-champ weights first
+
+  std::swap(*data.cm->_gd_normalized, data.cm->per_live_model_state_double[0]);
+  std::swap(*data.cm->_gd_total_weight, data.cm->per_live_model_state_double[1]);
+  std::swap(*data.cm->_sd_gravity, data.cm->per_live_model_state_double[2]);
+  std::swap(*data.cm->_cb_adf_event_sum, data.cm->per_live_model_state_uint64[0]);
+  std::swap(*data.cm->_cb_adf_action_sum, data.cm->per_live_model_state_uint64[1]);
+
+  clear_non_champ_weights(data.cm->weights, data.cm->estimators.size(), data.cm->wpp);
```

Comment:
could you write a comment here perhaps explaining what the code below is attempting to do?

Reply:
added comments for each operation